jeeves/src/main/java/jeeves/utils/XmlRequest.java

```diff
@@ -102,6 +95,10 @@ public XmlRequest(String host, int port, String protocol)
 		authPrefs.add(AuthPolicy.BASIC); 
 		// This will exclude the NTLM authentication scheme 
 		client.getParams().setParameter(AuthPolicy.AUTH_SCHEME_PRIORITY, authPrefs);
+		// Set default timeouts
+		HttpConnectionManagerParams defaultParams = client.getHttpConnectionManager().getParams();
+		defaultParams.setConnectionTimeout(5 * MILLISECONDS_IN_A_MINUTE);
+		defaultParams.setSoTimeout(5 * MILLISECONDS_IN_A_MINUTE);
```

Comment:
Would it make sense to set the Http timeout to be slightly shorter than the socket timeout. This would permit more graceful Http termination, using HTTP/1.1 Connection: close if possible.